fairseq/tasks/speech_to_speech.py

```diff
@@ -208,15 +213,28 @@ def __init__(self, args, tgt_dict, infer_tgt_lang_id=None):
         super().__init__(args)
         self.tgt_dict = tgt_dict
         self.data_cfg = S2SDataConfig(Path(args.data) / args.config_yaml)
+
         self.multitask_tasks = {}
+        self.tgt_dict_mt = None
+        self.eos_token_mt = None
         if getattr(args, "multitask_config_yaml", None) is not None:
             multitask_cfg = MultitaskConfig(
                 Path(args.data) / args.multitask_config_yaml
             )
             for task_name, task_config in multitask_cfg.get_all_tasks().items():
-                self.multitask_tasks[task_name] = DummyMultiTask(
-                    task_config, task_config.tgt_dict
-                )
+                task_obj = DummyMultiTask(task_config, task_config.tgt_dict)
+                self.multitask_tasks[task_name] = task_obj
+                if "target" in task_name and task_obj.args.decoder_type != "ctc":
```

Comment:
Is this mainly for the first decoder target for dual decoder architecture? I wonder if we shouldn't solely rely on `"target" in task_name` but add other fields to mark this special type of multitask, since previously we can also have an encoder multitask named "target_letter", "target_phoneme", or anything we'd like. Having an additional indicator for first decoder target will also allow us not to rely on the assumption that the last task is for the first-pass decoder.

Reply:
@an918tw Thank you for the useful feedback! I added a flag of "is_first_pass_decoder" in the multitask config file. If it is not specified, the last non-CTC decoder will be used as a first-pass decoder.